packages/botonic-api/src/rest/routes/users.ts

```diff
@@ -15,113 +15,82 @@ import {
   userWithUserIdParamSchema,
 } from './validation/users'
 
-const router = Router()
-
-router
-  .route('/')
-  .get(
-    checkSchema({
-      page: pageParamSchema,
-      pageSize: pageSizeParamSchema,
-    }),
-    async (req, res) => {
-      try {
-        const errors = validationResult(req)
-        if (!errors.isEmpty()) {
-          res.status(400).send({ errors: errors.array() })
-          return
+export default function usersRouter(args) {
+  const router = Router()
+  router
+    .route('/')
+    .get(
+      checkSchema({
+        page: pageParamSchema,
+        pageSize: pageSizeParamSchema,
+      }),
+      async (req, res) => {
+        try {
+          const errors = validationResult(req)
+          if (!errors.isEmpty()) {
+            res.status(400).send({ errors: errors.array() })
+            return
+          }
+
+          const query = matchedData(req, { locations: ['query'] })
+          const paginator = new Paginator(req, query.page, query.pageSize)
+          const dp = dataProviderFactory(process.env.DATA_PROVIDER_URL)
```

Comment:
maybe we could allow to call `dataProviderFactory` without params and the function will try to get the URL from the standard env var, like:
```
function dataProviderFactory(url) {
  if(!url) url = process.env.DATA_PROVIDER_URL
  ...
}
```